src/vuln_analysis/runtime_context.py

```diff
@@ -0,0 +1,10 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# A shared runtime context for passing workflow state to tools and functions.
+# This must be imported by all modules that need to read or set the active state.
+
+import contextvars
+
+# Holds the current AgentMorpheusEngineState for the active task
+ctx_state = contextvars.ContextVar("ctx_state", default="default_value")
```

Comment:
@IlonaShishov After this change, The following will break:
https://github.com/RHEcosystemAppEng/vulnerability-analysis/blob/ee9a733de5ebcd6bcafe8fe18940051e81a4c13d/src/vuln_analysis/tools/transitive_code_search.py#L16
Please change the import there,  according to this new module.